### PR TITLE
Update to valid params and remove PresetBC from tests

### LINKED ISSUE #24: Update Deer InputParameters Syntax
we need to update deer to the new InputParameters syntax.
We also need to remove all PresetBC in the tests as they are deprecated. The new standard is DirichletBC with `preset=true`

### PR DESCRIPTION
**Pull request by arovinelli**
we need to update deer to the new InputParameters syntax.
We also need to remove all PresetBC in the tests as they are deprecated. The new standard is DirichletBC with preset=true

closes #24

### CODE REVIEW

File: examples/euler_angle-reader/cpfem_eulerTest.i [hunk 1]
```diff
@@ -102,21 +102,24 @@
...
     value = 0.0
   [../]
 
   [./back]
-    type = PresetBC
+    type = DirichletBC
+     preset = true
     variable = disp_z
     boundary = back
     value = 0.0
```
Comment (chakra34) [inline examples/euler_angle-reader/cpfem_eulerTest.i]: the last boundary condition needs to be changed to FunctionDirichletBC as well.

File: examples/euler_angle-reader/small_eulerTest.i [hunk 2]
```diff
@@ -25,19 +25,22 @@
...
     variable = disp_y
     value = 0.0
   [../]
   [./leftz]
-    type = PresetBC
+    type = DirichletBC
+    preset = true
     boundary = bottom
     variable = disp_z
     value = 0.0
```
Comment (chakra34) [inline examples/euler_angle-reader/small_eulerTest.i]: same as above

**Comment by arovinelli** [comment]
changed `FunctionPresetBC` to `FunctionDiricheltBC`
should be good to go now

**Comment by arovinelli** [comment]
@chakra34 @reverendbedford if anyone approves we can merge

**Pull request merged**